runtime/src/tiered_storage/hot.rs

```diff
@@ -199,6 +202,39 @@ impl TieredAccountMeta for HotAccountMeta {
     }
 }
 
+/// The reader to a hot accounts file.
+#[derive(Debug)]
+pub struct HotStorageReader {
+    map: Mmap,
+    footer: TieredStorageFooter,
+}
+
+impl HotStorageReader {
+    /// Constructs a HotStorageReader from the specified path.
+    pub fn new_from_path<P: AsRef<Path>>(path: P) -> TieredStorageResult<Self> {
+        let file = OpenOptions::new()
+            .read(true)
+            .create(false)
+            .open(path.as_ref())?;
+        let map = unsafe { MmapOptions::new().map(&file)? };
+        let footer = TieredStorageFooter::new_from_mmap(&map)?.clone();
```

Comment:
Because it returns `&` instead of an instance.  Without cloning, it will require the Reader to have `<'a>`.  Or, is there a workaround to make the Reader smart enough that it is from the mmap itself?

Reply:
Thanks for the information.  Let's leave it in a potential future PR.

Reply:
Added comments to the code.